pkg/ytconfig/node.go

```diff
@@ -325,7 +325,8 @@ func getDataNodeServerCarcass(spec *ytv1.DataNodesSpec) (DataNodeServer, error)
 		}
 		if quota != nil {
 			storeLocation.Quota = *quota
-
+		}
+		if location.LowWatermark != nil or storeLocation.Quota != nil {
```

Comment:
Let's also move `storeLocation.MaxTrashTtl = location.MaxTrashMilliseconds` out of this `if`.